src/snowflake/cli/plugins/object/manager.py

```diff
@@ -66,3 +69,85 @@ def object_exists(self, *, object_type: str, name: str):
             return True
         except ProgrammingError:
             return False
+
+    def _send_rest_request(
+        self, url: str, method: str, data: Optional[Dict[str, Any]] = None
+    ):
+        # SnowflakeRestful.request assumes that API response is always a dict,
+        # which is not true in case of this API, so we need to do this workaround:
+        from snowflake.connector.network import (
+            CONTENT_TYPE_APPLICATION_JSON,
+            HTTP_HEADER_ACCEPT,
+            HTTP_HEADER_CONTENT_TYPE,
+            HTTP_HEADER_USER_AGENT,
+            PYTHON_CONNECTOR_USER_AGENT,
+        )
+
+        self._log.debug(f"Sending {method} request to {url}")
+        rest = self._conn.rest
+        full_url = f"{rest.server_url}{url}"
+        headers = {
+            HTTP_HEADER_CONTENT_TYPE: CONTENT_TYPE_APPLICATION_JSON,
+            HTTP_HEADER_ACCEPT: CONTENT_TYPE_APPLICATION_JSON,
+            HTTP_HEADER_USER_AGENT: PYTHON_CONNECTOR_USER_AGENT,
+        }
+        return rest.fetch(
+            method=method,
+            full_url=full_url,
+            headers=headers,
+            token=rest.token,
+            data=json.dumps(data if data else {}),
+            no_retry=True,
+        )
+
+    def _url_exists(self, url):
+        try:
+            result = self._send_rest_request(url, method="get")
+            return bool(result) or result == []
+        except InterfaceError as err:
+            if "404 Not Found" in str(err):
+                return False
+            raise err
+
+    def _get_rest_api_create_url(self, object_type: str):
+        if object_type.endswith("y"):
+            plural_object_type = object_type[:-1].lower() + "ies"
+        else:
+            plural_object_type = object_type.lower() + "s"
+
+        url_prefix = "/api/v2"
+
+        url = f"{url_prefix}/{plural_object_type}/"
+        if self._url_exists(url):
+            return url
```

Comment:
Add comments explaining why this pre-check

Reply:
Added a docstring and refactored so it is more clear